skimage/restoration/inpaint.py

```diff
@@ -126,25 +252,88 @@ def inpaint_biharmonic(image, mask, multichannel=False):
     image = skimage.img_as_float(image)
     mask = mask.astype(bool)
 
-    # Split inpainting mask into independent regions
-    kernel = ndi.morphology.generate_binary_structure(mask.ndim, 1)
-    mask_dilated = ndi.morphology.binary_dilation(mask, structure=kernel)
-    mask_labeled, num_labels = label(mask_dilated, return_num=True)
-    mask_labeled *= mask
-
     if not multichannel:
         image = image[..., np.newaxis]
-
     out = np.copy(image)
 
-    for idx_channel in range(image.shape[-1]):
-        known_points = image[..., idx_channel][~mask]
-        limits = (np.min(known_points), np.max(known_points))
-
-        for idx_region in range(1, num_labels+1):
-            mask_region = mask_labeled == idx_region
-            _inpaint_biharmonic_single_channel(mask_region,
-                out[..., idx_channel], limits)
+    # Create biharmonic coefficients ndarray
+    radius = 2
+    coef_shape = (2 * radius + 1,) * mask.ndim
+    coef_center = (radius,) * mask.ndim
+    neigh_coef_full, coef_idx, coef_vals = _get_neigh_coef(coef_shape,
+                                                           coef_center,
+                                                           coef_cache={},
```

Comment:
```suggestion
```

Reply:
Most likely I should just not use an empty dict in the function signature for `_get_neigh_coef` and default to `coef_cache=None` instead). Otherwise the dict persists across function calls and passing an empty dict like this was needed to avoid side effects!

Reply:
You are right, simply not defining a default value for `coef_cache` may do the trick...